src/ValidatorFactory.php

```diff
@@ -175,7 +178,34 @@ public static function length(Input $input): Constraint
             $options += self::options($input, 'maxlength', ['max' => $maxlength], 'maxMessage');
         }
 
-        return new Constraints\Length($options);
+        switch (self::getValidatorVersion()) {
+            case 7:
+                return new \FormManager\Constraints\OptionalLength7($options);
+            case 6:
+                return new \FormManager\Constraints\OptionalLength6($options);
+            case 5:
+                $options['allowEmptyString'] = true;
+                return new Constraints\Length($options);
+        }
+    }
+
+    private static function getValidatorVersion(): int
+    {
+        if (property_exists(Constraints\Length::class, 'allowEmptyString')) {
+            return 5;
+        }
+
+        $reflection = new ReflectionClass(LengthValidator::class);
+        $hasReturnType = $reflection->getMethod('validate')->hasReturnType();
+        if (!$hasReturnType) {
+            return 6;
+        }
+
+        if ($hasReturnType) {
+            return 7;
+        }
+
+        throw new RuntimeException('Cannot determine symfony/validator version. Please report this on GitHub.');
```

Comment:
interesting line here @oscarotero. If we put a fallback it will fail with a weid error. (PHP native errors. So I thought this is the best to communicate what's going on.